lib/features/support/interactive_cli.rb

```diff
@@ -0,0 +1,117 @@
+require 'open3'
+
+# Encapsulates a terminal session, retaining state and input streams for interactive tests
+class InteractiveCLI
+  # @!attribute [r] parsed_output
+  #   @return [Array] An array of output strings received from the terminals STDOUT pipe
+  attr_reader :parsed_output
+
+  # @!attribute [r] parsed_error
+  #   @return [Array] An array of error strings received from the terminals STDERR pipe
+  attr_reader :parsed_errors
+
+  # @!attribute [r] last_exit_code
+  #   @return [Number] The exit code of the last terminal command
```

Comment:
This will be `nil` if a command hasn't been run

```suggestion
  #   @return [Number, nil] The exit code of the last terminal command
```

Reply:
Interesting - I wasn't aware of the ability/convention to explicitly not the possibility of `nil`s.